tmt/steps/prepare/install.py

```diff
@@ -134,6 +156,41 @@ def enable_copr(self, command, plugin, guest):
                 self.info('copr', copr, 'green')
                 guest.execute(f'{command} copr enable -y {copr}')
 
+    def enable_custom_repo(self, command, guest):
+        """ Enable requested repositories """
+        repos = self.get('repo')
+        if not repos:
+            return
+
+        # Create repo file content one bye one
+        repo_configs = list()
+        for repo_id, repo in enumerate(repos):
+            self.debug(f'Create repo file content for {repo}.')
+            repo_name = f'custom-{repo_id}'
+            repo_config = configparser.ConfigParser()
+            repo_config.add_section(repo_name)
+            repo_config.set(repo_name, 'name', repo_name)
+            repo_config.set(repo_name, 'baseurl', repo)
+            repo_config.set(repo_name, 'enable', '1')
+            repo_config.set(repo_name, 'gpgcheck',
+                            '1' if self.get('gpgcheck') else '0')
+            repo_config.set(repo_name, 'skip_if_unavailable',
+                            '1' if self.get('skip-if-unavailable')
+                            else '0')
+            repo_configs.append(repo_config)
+
+        # Try to save repo files to guest
+        self.debug('Save repo file to guest.')
+        for repo_config in repo_configs:
```

Comment:
I'm curious, why the second loop? What is the advantage of creating repo configurations in memory first? I don't see any issue in creating the configuration & pushing it to remote immediately, and all variables would be available already.

Reply:
Hi @happz, what you say is correct, we don't have to have the second loop. The related code block is updated to use one loop only, please help to take a look again, thanks!

Reply:
Lovely, thanks :)